common/persistence/sql/blob.go

```diff
@@ -38,59 +42,103 @@ type thriftRWType interface {
 	FromWire(w wire.Value) error
 }
 
-func validateProto(p string) error {
-	if common.EncodingType(p) != common.EncodingTypeThriftRW {
+type protoMarshal interface {
+	Marshal() ([]byte, error)
+	Unmarshal([]byte) error
+}
+
+func validateProto(p string, expected common.EncodingType) error {
+	if common.EncodingType(p) != expected {
 		return fmt.Errorf("invalid encoding type: %v", p)
 	}
 	return nil
 }
 
-func encodeErr(err error) error {
+func encodeErr(encoding common.EncodingType, err error) error {
 	if err == nil {
 		return nil
 	}
-	return fmt.Errorf("error serializing struct to thrift blob: %v", err)
+	return fmt.Errorf("error serializing struct to blob using encoding - %v - : %v", encoding, err)
 }
 
-func decodeErr(err error) error {
+func decodeErr(encoding common.EncodingType, err error) error {
 	if err == nil {
 		return nil
 	}
-	return fmt.Errorf("error deserializing blob to thrift struct: %v", err)
+	return fmt.Errorf("error deserializing blob to blob using encoding - %v - : %v", encoding, err)
+}
+
+func protoRWEncode(m protoMarshal) (p.DataBlob, error) {
```

Comment:
I don't like `p` alias here. Can you remove it?

Reply:
ok, may be later.

Reply:
p for persistence is pretty highly used throughout the code, we'll need to do a bunch of renaming rounds after these refactors